src/main/java/org/openrewrite/java/spring/boot2/WebSecurityConfigurerAdapterRecipe.java

```diff
@@ -0,0 +1,240 @@
+                        maybeAddImport(FQN_WEB_SECURITY_CUSTOMIZER);
+                    } else if (CONFIGURE_AUTH_MANAGER_SECURITY_METHOD_MATCHER.matches(m, classCursor.getValue())) {
+                        // TODO: implement this case
+                        m = m.withMarkers(m.getMarkers().searchResult("Migrate manually based on https://spring.io/blog/2022/02/21/spring-security-without-the-websecurityconfigureradapter"));
+                    }
+                }
+                return m;
+            }
+
+            @Override
+            public J.Block visitBlock(J.Block block, ExecutionContext context) {
+                J.Block b = super.visitBlock(block, context);
+                if (getCursor().getParent() != null && getCursor().getParent().getValue() instanceof J.MethodDeclaration) {
+                    J.MethodDeclaration parentMethod = getCursor().getParent().getValue();
+                    Cursor classDeclCursor = getCursor().dropParentUntil(J.ClassDeclaration.class::isInstance);
+                    J.ClassDeclaration classDecl = classDeclCursor.getValue();
+                    if (classDeclCursor.getMessage(MSG_QUALIFIES, false)) {
+                        if (CONFIGURE_HTTP_SECURITY_METHOD_MATCHER.matches(parentMethod, classDecl)) {
+                            J.VariableDeclarations.NamedVariable paramDef = ((J.VariableDeclarations) parentMethod.getParameters().get(0)).getVariables().get(0);
+                            J.Identifier methodName = new J.Identifier(Tree.randomId(), Space.EMPTY, Markers.EMPTY, "build", null, null);
+                            JavaType.Method methodType = TypeUtils.findDeclaredMethod(TypeUtils.asFullyQualified(paramDef.getType()), "build", Collections.emptyList())
+                                    .orElse(new JavaType.Method(null, 0, null, "build", null, null, null, null, null));
+                            J.Identifier variableIdentifier = new J.Identifier(Tree.randomId(), Space.EMPTY, Markers.EMPTY, paramDef.getSimpleName(), paramDef.getType(), null);
+                            J.MethodInvocation invocation = new J.MethodInvocation(Tree.randomId(), Space.build(" ", Collections.emptyList()), Markers.EMPTY, JRightPadded.build(variableIdentifier), null, methodName, JContainer.build(Collections.emptyList()), methodType);
+                            J.Return returnStatement = new J.Return(Tree.randomId(), Space.build("\n", Collections.emptyList()), Markers.EMPTY, invocation);
+                            b = b.withStatements(ListUtils.concat(b.getStatements(), autoFormat(returnStatement, context)));
+//                        JavaTemplate template = JavaTemplate.builder(() -> getCursor(), "return " + ((J.VariableDeclarations)parentMethod.getParameters().get(0)).getVariables().get(0).getSimpleName() + ".build();").javaParser(() -> JavaParser.fromJavaVersion().build()).build();
+//                        b = b.withTemplate(template, b.getCoordinates().addStatement(null));
```

Comment:
your JavaTemplate was super close it just needed a parameter reference and a symbol for the SecurityBuilder.
check this out:

```java
JavaTemplate template = JavaTemplate.builder(this::getCursor,  "return #{any(org.springframework.security.config.annotation.SecurityBuilder)}.build();")
                                   .javaParser(() -> JavaParser.fromJavaVersion()
                                           .dependsOn("package org.springframework.security.config.annotation;" +
                                                   "public interface SecurityBuilder<O> {\n" +
                                                   "    O build() throws Exception;" +
                                                   "}")
                                           .build()).imports("org.springframework.security.config.annotation.SecurityBuilder").build();
                            b = b.withTemplate(template, b.getCoordinates().lastStatement(), ((J.VariableDeclarations)parentMethod.getParameters().get(0)).getVariables().get(0).getName());
```